### DISCUSSION TITLE
SubMenu with `overflow: visible` has overflow issue when using many MenuItem

### DISCUSSION THREAD

**Discussion by michaelsawyers** [ariakit/ariakit#3537] [Questions] [answered]
For the [SubMenu example](https://ariakit.org/examples/menu-nested) when there are enough MenuItems to surpass the `--popover-available-height`, how can I prevent the MenuItems from overflowing? 
I would like to see scroll behavior, but due to the `overflow: visible` set on Menu the MenuItems overflow. 
I'm aware that without the `overflow: visible` property the submenus are hidden.

Stumpted on this, anyone got any good ideas?
Forked from SubMenu example with added MenuItems: https://stackblitz.com/edit/sjvhtc?file=menu-nested%2Findex.tsx

**Comment by diegohaz** [comment, marked as answer]
You can use `overflow: auto`, but the submenu must be rendered outside the parent menu, either by rendering it outside in the React tree or by using the `portal` prop on the sub menu component.

See [Menubar](https://ariakit.org/components/menubar).

**Reply by michaelsawyers** [reply]
Thanks @diegohaz this is what I'm looking for
